lightning-net-tokio/src/lib.rs

```diff
@@ -450,18 +437,10 @@ impl peer_handler::SocketDescriptor for SocketDescriptor {
 	}
 
 	fn disconnect_socket(&mut self) {
-		{
-			let mut us = self.conn.lock().unwrap();
-			us.rl_requested_disconnect = true;
-			us.read_paused = true;
```

Comment:
Why no need to set `read_paused` anymore?

Reply:
Because we no longer need to avoid calling `read_event` at all costs - it won't panic after this point but will instead simply return an error.